src/scorepy/cInstrumenter.cpp

```diff
@@ -113,33 +114,32 @@ bool CInstrumenter::on_event(PyFrameObject& frame, int what, PyObject*)
     {
     case PyTrace_CALL:
     {
-        const PyCodeObject& code = *frame.f_code;
-        const char* name = PyUnicode_AsUTF8(code.co_name);
+        const PyCodeObject* code = frame.f_code;
```

Comment:
I actually wanted to use a reference here to signal "cannot be NULL". Would it make sense to leave that and change the below to `reinterpret_cast<std::uintptr_t>(&code)`?

Reply:
Interesting convention.
But I would prefer to use a reference when we want to deal with a reference and a pointer when we deal with a pointer. As we are doing later, I would like to stay with that here.

Reply:
It is actually a bit more than just a convention. Foremost: It is undefined behavior for a reference to be `"NULL"`, so this is a "language convention" if you want.

But my point is: (Raw) pointers are hard. You get into a whole territory of issues/questions. The first 2: Can this be NULL? And who owns this / do I need to delete this?   
Consistently using smart pointers eliminates the 2nd question as a raw pointer will then never be owning (by convention as it would otherwise be a smart pointer) so the first one remains: Can this be NULL?

So I'd avoid raw pointers unless they actually can be NULL and use references instead.

Reply:
Usually, I would agree. However, as python already uses Pointer for everything, and the whole story of ownership and "can this be NULL" is already an issue, I would prefer to stay with raw pointers.

However, if you insist, I am ok with changing this here.

Reply:
>  However, as python already uses Pointer for everything, and the whole story of ownership and "can this be NULL" is already an issue, I would prefer to stay with raw pointers.

Yes we are using a C interface and as C has neither references nor smart pointers the best you can do is to "translate" at the interface level to mitigate the problem and avoid it in the "fully C++" level. This is done by wrapping pointers into smart-pointers or similar RAII classes like I did with https://github.com/score-p/scorep_binding_python/blob/6a766ae3fe716d0ea9fdc4c6ca4278cf09aacf7e/src/scorepy/pythonHelpers.hpp#L24 for "owned references/instances/pointers".

So I'd rather go further into this direction and get rid of any raw pointer except where it really means "an optional reference to a non-owned object". Obviously you can't do that when calling a C function, but there you can go from reference to pointer easily. The other way round might not always be possible, e.g. where a C function has a pointer as an out-parameter. But that should then be "translated" to C++ ASAP. 

I hope this makes the motivation clear. Also I'm not proposing to do this change on the whole codebase now but rather a goal to "fix-as-you-go" and also to not introduce "regressions" like here.

Reply:
done